basic_features/basic_mod_data_checker.py

```diff
@@ -0,0 +1,258 @@
+    @overload
+    def __init__(
+        self,
+        *,
+        unfold: Iterable[str] | None = None,
+        valid: Iterable[str] | None = None,
+        delete: Iterable[str] | None = None,
+        move: dict[str, str] | None = None,
+    ):
+        ...
+
+    @overload
+    def __init__(
+        self,
+        file_patterns: Optional[GlobPatternDict] = None,
+        *,
+        unfold: Iterable[str] | None = None,
+        valid: Iterable[str] | None = None,
+        delete: Iterable[str] | None = None,
+        move: dict[str, str] | None = None,
+    ):
+        ...
+
+    def __init__(
+        self, file_patterns: Optional[GlobPatternDict] = None, **kwargs
+    ):  # Unpack[GlobPatternDict]
+        super().__init__()
+        # Init with copy from class var
+        self.file_patterns = fp.copy() if (fp := self.file_patterns) else {}
+        self.update_patterns(file_patterns, **kwargs)
+
+    @overload
+    def update_patterns(
+        self,
+        *,
+        unfold: Iterable[str] | None = None,
+        valid: Iterable[str] | None = None,
+        delete: Iterable[str] | None = None,
+        move: dict[str, str] | None = None,
+    ):
+        ...
+
+    @overload
+    def update_patterns(
+        self,
+        file_patterns: Optional[GlobPatternDict] = None,
+        *,
+        unfold: Iterable[str] | None = None,
+        valid: Iterable[str] | None = None,
+        delete: Iterable[str] | None = None,
+        move: dict[str, str] | None = None,
+    ):
+        ...
+
+    def update_patterns(self, file_patterns=None, **kwargs):  # Unpack[GlobPatternDict]
+        """Update file patterns."""
+        if file_patterns:
+            self.file_patterns.update(file_patterns)
+        self.file_patterns.update(kwargs)
+        self._regex = RegexPatternDict.from_glob_patterns(self.file_patterns)
+        if move_map := self.file_patterns.get("move"):
+            self._move_targets = list(move_map.values())
+
+    def dataLooksValid(
+        self, filetree: mobase.IFileTree
+    ) -> mobase.ModDataChecker.CheckReturn:
+        status = mobase.ModDataChecker.INVALID
+        for entry in filetree:
+            name = entry.name().casefold()
+            for key, regex in self._regex.items():
+                if not regex or not regex.match(name):
+                    continue
+                if key == "unfold":
+                    return mobase.ModDataChecker.FIXABLE
+                elif key == "valid":
+                    if status is not mobase.ModDataChecker.FIXABLE:
+                        status = mobase.ModDataChecker.VALID
+                elif key in ("move", "delete"):
+                    status = mobase.ModDataChecker.FIXABLE
+                break
+            else:
+                return mobase.ModDataChecker.INVALID
+        return status
+
+    def fix(self, filetree: mobase.IFileTree) -> Optional[mobase.IFileTree]:
+        print(f"{self.file_patterns}")
```

Comment:
These print should be removed or moved to the debug logs.

Reply:
True, missed one, thanks!